Source/WebCore/fileapi/Blob.cpp

```diff
@@ -55,21 +57,56 @@ WTF_MAKE_ISO_ALLOCATED_IMPL(Blob);
 
 class BlobURLRegistry final : public URLRegistry {
 public:
-    void registerURL(ScriptExecutionContext&, const URL&, URLRegistrable&) final;
-    void unregisterURL(const URL&) final;
+    void registerURL(const ScriptExecutionContext&, const URL&, URLRegistrable&) final;
+    bool unregisterURL(const URL&) final;
+    void unregisterURLsForContext(const ScriptExecutionContext&) final;
 
     static URLRegistry& registry();
+
+    Lock m_urlsPerContextLock;
+    HashMap<ScriptExecutionContextIdentifier, HashSet<URL>> m_urlsPerContext WTF_GUARDED_BY_LOCK(m_urlsPerContextLock);
 };
 
-void BlobURLRegistry::registerURL(ScriptExecutionContext& context, const URL& publicURL, URLRegistrable& blob)
+void BlobURLRegistry::registerURL(const ScriptExecutionContext& context, const URL& publicURL, URLRegistrable& blob)
 {
     ASSERT(&blob.registry() == this);
+    {
+        Locker locker { m_urlsPerContextLock };
+        m_urlsPerContext.add(context.identifier(), HashSet<URL>()).iterator->value.add(publicURL.isolatedCopy());
+    }
     ThreadableBlobRegistry::registerBlobURL(context.securityOrigin(), context.policyContainer(), publicURL, static_cast<Blob&>(blob).url());
 }
 
-void BlobURLRegistry::unregisterURL(const URL& url)
+bool BlobURLRegistry::unregisterURL(const URL& url)
 {
+    bool isURLRegistered = false;
```

Comment:
`wasURLRegistered`

Reply:
Maybe it's my bad English but `wasURLRegistered` sounds a little more confusing to me than `isURLRegistered` in this context. First we check if the URL *is* registered, and if it is registered then I unregister it. 

If this were a registerURL() function and I'd want a boolean to keep track of whether or not the function actually registered the URL, then I would use `wasURLRegistered` because it would convey: "Did the earlier code register the URL".

As I said, it may just be my English.